waspc/cli/src/Wasp/Cli/Command/Compile.hs

```diff
@@ -24,13 +25,18 @@ import qualified Wasp.Message as Msg
 
 compile :: Command ()
 compile = do
+  waspProjectDir <- findWaspProjectRootDirFromCwd
```

Comment:
Yeah, it is not ideal, but I'm not too sure of a clean way to avoid it twice. I see a few options:
- Try to pull it out of `externalCodeDirPath` but that is just duplication in the opposite direction :D
- Add a new field to `CompileOptions` just for `waspProjectDir`, and then just add helpers for `externalCodeDirPath` and maybe even the generated out dir path (or whatever else others need)
- Pass it into `compileWithOptions`, but now that puts a new burden on all callers, which is prob worse

Option 2 seems likely the best, but I wonder if we should tackle that refactor right now? My inclination is probably to pay the redundancy price in this PR. Can TODO it so we do not forget option 2 though.